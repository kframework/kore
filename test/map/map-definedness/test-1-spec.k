module VERIFICATION
  imports TEST
endmodule

module TEST-1-SPEC
  imports VERIFICATION

  rule
    <k> add(X, 3) => end </k>
    <a> Y:Int |-> 2 M:Map => ?_ </a>

endmodule
