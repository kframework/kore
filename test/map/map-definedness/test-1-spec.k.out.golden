  #Not ( #Not ( {
      X
    #Equals
      Y
    } )
  #And
    {
      false
    #Equals
      X:Int in_keys ( M )
    } )
#And
  <T>
    <k>
      added
    </k>
    <a>
      M
      X:Int |-> 3
      Y:Int |-> 2
    </a>
  </T>
#And
  {
    Y:Int in_keys ( M )
  #Equals
    false
  }
